Tests/OTelTests/OTelCoreTests/Logging/DiagnosticLoggingTests.swift

```diff
@@ -0,0 +1,36 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2025 the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Logging
+@testable import OTel
+import Testing
+
+@Suite struct DiagnosticLoggingTests {
+    @Test func testLogErrorHelper() async throws {
+        let recordingHandler = RecordingLogHandler()
+        var logMessages = recordingHandler.recordedLogMessageStream.makeAsyncIterator()
+        let logger = Logger(label: "test") { _ in recordingHandler }
+        struct TestError: Error, CustomStringConvertible {
+            var description: String { "custom description" }
+        }
+
+        for level in [Logger.Level.trace, .trace, .info, .warning, .error] {
```

Comment:
Nitpick: How do you feel about using a parameterized test instead of an inner for loop? I usually try to go with them instead because the test report looks slightly better IMHO.

Reply:
Sure, the previous time I tried to use them they weren't working because of limitations with the exit tests. But you're right: this is a perfect use case.